_extensions/custom_roles.py

```diff
@@ -1,43 +0,0 @@
-"""Sphinx extension to add custom roles.
-
-Based on https://protips.readthedocs.io/link-roles.html
-"""
-import urllib.parse
-
-from docutils import nodes
-
-
-def setup(app):
-    # role to link to cpython files
-    app.add_role(
-        "cpy-file",
-        autolink("https://github.com/python/cpython/blob/main/{}"),
-    )
-    # role to link to cpython labels
-    app.add_role(
-        "gh-label",
-        autolink("https://github.com/python/cpython/labels/{}"),
-    )
-    # Parallel safety:
-    # https://www.sphinx-doc.org/en/master/extdev/index.html#extension-metadata
-    return {"parallel_read_safe": True, "parallel_write_safe": True}
-
-
-def autolink(pattern):
-    def role(name, rawtext, text, lineno, inliner, _options=None, _content=None):
-        """Combine literal + reference (unless the text is prefixed by a !)."""
-        if " " in text:
-            url_text = urllib.parse.quote(text)
-        else:
-            url_text = text
-        url = pattern.format(url_text)
-        # don't create a reference if the text starts with !
-        if text.startswith('!'):
```

Comment:
Does this work with extlinks?

Reply:
No; see https://cpython-devguide--1351.org.readthedocs.build/internals/parser/#how-to-regenerate-the-parser

However we could just use ``:file:`Makefile` `` or ``` ``Makefile`` ```?

A

Reply:
We have quite a few ``` ``Makefile`` ``` already, so I've changed ``` :cpy-file:`!Makefile` ``` to that.